<?xml version="1.0"?>
<launch>
    <!-- number of moving objects -->
    <arg name="obj_num" value="10" />
    <arg name="drone_id" value="0"/>

    <arg name="map_size_x" value="20.0"/>
    <arg name="map_size_y" value="10.0"/>
    <arg name="map_size_z" value=" 4.0"/>
    <arg name="odom_topic" value="visual_slam/odom" />
    
    <!-- map -->
    <node pkg ="map_generator" name ="map_generator" type ="random_env" output = "screen">
        <param name="sensing/rate"   value="10.0"/>   
        <param name="map/x_size"     value="$(arg map_size_x)" />
        <param name="map/y_size"     value="$(arg map_size_y)" />
        <param name="map/z_size"     value="$(arg map_size_z)" />
        <param name="map/resolution" value="0.1"/>        
        <param name="map/frame_id"     value="world" />
        <param name="map/seed" value="1"/>
        <param name="map/obs_num"    value="20"/>

        <param name="map/future_num"    value="9"/>
        <param name="map/time_step"    value="0.2"/>
    
        <param name="obstacle/upper_vel" value="1.0"/>

        <param name="aabb/num" value="15"/>
        <param name="aabb/lower_x" value="0.3"/>
        <param name="aabb/upper_x" value="2.0"/>
        <param name="aabb/lower_y" value="0.3"/>
        <param name="aabb/upper_y" value="2.5"/>        
        <param name="aabb/lower_z" value="0.3"/>
        <param name="aabb/upper_z" value="2.5"/>        

        <param name="cylinder/num" value="15"/>
        <param name="cylinder/lower_rad" value="0.2"/>
        <param name="cylinder/upper_rad" value="0.5"/>
        <param name="cylinder/lower_hei" value="4.0"/>
        <param name="cylinder/upper_hei" value="4.0"/>        
    
        <param name="circlegate/num" value="7"/>        
        <param name="circlegate/radius_l" value="0.7"/>        
        <param name="circlegate/radius_h" value="2.5"/>        
        <param name="circlegate/thickness" value="0.1"/>        
        <param name="circlegate/dr" value="0.3"/>        
        <param name="circlegate/theta" value="1.2"/>        

        <param name="pcd/num" value="2" />
        <param name="pcd/path" value="$(find map_generator)/data/text.pcd" />
    </node>

  <!-- fake camera sensing -->
    <node pkg="map_generator" type="pcl_render_node" name="pcl_render_node" output="screen">
        <param name="sensing_horizon" value="6.0" />
        <param name="sensing_rate" value="30.0" />
        <param name="estimation_rate" value="30.0" />
        <param name="is_camera_frame" value="true" />
        <param name="is_dynamic_map" value="true" />

        <param name="global_map/x_size" value="$(arg map_size_x)" />
        <param name="global_map/y_size" value="$(arg map_size_y)" />
        <param name="global_map/z_size" value="$(arg map_size_z)" />
        <param name="global_map/resolution" value="0.1" />

        <remap from="~future_risk" to="/map_generator/future_risk" />
        <remap from="~global_map" to="/map_generator/global_cloud" />
        <remap from="~odometry" to="visual_slam/odom" />
        <remap from="~pcl_render_node/cloud" to="~/cloud" />
    </node>

    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find map_generator)/launch/default.rviz" required="true" />
      
</launch>
